recipes/qt/6.x.x/conanfile.py

```diff
@@ -0,0 +1,543 @@
+        if not self.options.gui:
+            del self.options.opengl
+            del self.options.with_vulkan
+            del self.options.with_freetype
+            del self.options.with_fontconfig
+            del self.options.with_harfbuzz
+            del self.options.with_libjpeg
+            del self.options.with_libpng
+
+        if self.settings.os == "Android" and self.options.get_safe("opengl", "no") == "desktop":
+            raise ConanInvalidConfiguration("OpenGL desktop is not supported on Android.")
+
+        if self.settings.os != "Windows" and self.options.get_safe("opengl", "no") == "dynamic":
+            raise ConanInvalidConfiguration("Dynamic OpenGL is supported only on Windows.")
+
+        if self.options.get_safe("with_fontconfig", False) and not self.options.get_safe("with_freetype", False):
+            raise ConanInvalidConfiguration("with_fontconfig cannot be enabled if with_freetype is disabled.")
+
+        if self.options.multiconfiguration:
+            del self.settings.build_type
+
+        if "MT" in self.settings.get_safe("compiler.runtime", default="") and self.options.shared:
+            raise ConanInvalidConfiguration("Qt cannot be built as shared library with static runtime")
+
+        config = configparser.ConfigParser()
+        config.read(os.path.join(self.recipe_folder, "qtmodules%s.conf" % self.version))
+        submodules_tree = {}
+        assert config.sections()
+        for s in config.sections():
+            section = str(s)
+            assert section.startswith("submodule ")
+            assert section.count('"') == 2
+            modulename = section[section.find('"') + 1: section.rfind('"')]
+            status = str(config.get(section, "status"))
+            if status != "obsolete" and status != "ignore":
+                submodules_tree[modulename] = {"status": status,
+                                "path": str(config.get(section, "path")), "depends": []}
+                if config.has_option(section, "depends"):
+                    submodules_tree[modulename]["depends"] = [str(i) for i in config.get(section, "depends").split()]
+
+        for m in submodules_tree:
+            self.output.info(m)
+            assert(m == "qtbase" or m in self._submodules)
+
+        for m in self._submodules:
+            self.output.info(m)
+            assert(m in submodules_tree)
+
+        def _enablemodule(mod):
+            if mod != "qtbase":
+                setattr(self.options, mod, True)
+            for req in submodules_tree[mod]["depends"]:
+                _enablemodule(req)
+
+        for module in self._submodules:
+            if self.options.get_safe(module):
+                _enablemodule(module)
+
+    def requirements(self):
+        if self.options.openssl:
+            self.requires("openssl/1.1.1i")
+        if self.options.with_pcre2:
+            self.requires("pcre2/10.35")
+        if self.options.with_vulkan:
+            self.requires("vulkan-loader/1.2.162.0")
```

Comment:
How Qt6 uses vulkan loader? It it dynamically loads the function from shared lib, and doesn't include vulkan-headers, this requirement should be removed.

Reply:
https://doc-snapshots.qt.io/qt6-dev/qtgui-index.html#vulkan-integration

So it requires Vulkan SDK from LunarG, which is partially emulated in `vulkan-loader` recipe.

Reply:
I don't really understand what you are describing, but what I experienced is that [qt tries to find cmake module vulkan, and checks if it has a `Vulkan::Vulkan` target](https://code.qt.io/cgit/qt/qtbase.git/tree/src/gui/configure.cmake#n52).
I found out that [these cmake modules and target are provided by `vulkan-loader`](https://github.com/conan-io/conan-center-index/blob/master/recipes/vulkan-loader/all/conanfile.py#L138), and the build passes like this. What do you suggest to improve the situation ? (and what is the problem ?)

Reply:
Looking qtbase on github:
- vulkan-memoryallocator is required but vendored (and modified) https://github.com/qt/qtbase/blob/15576c961047231a49afda9b9ee0159ba132c0ae/src/gui/CMakeLists.txt#L272
- vulkan-headers are required, and they seem to highly depend on `VULKAN_SDK` env var to detect include dir.
- vulkan-loader is not required (only the dll/so at runime which is shipped in vulkan-loader, but expected to be provided with drivers), pointer functions are loaded dynamically: https://github.com/qt/qtbase/blob/3cb2b5bfe3e3c84dcd4c1a04c8c981dbaf85f0a4/src/gui/vulkan/qbasicvulkanplatforminstance.cpp
- moltenvk is required for macos (not yet available in CCI)

Reply:
quick notes:
1/ VULKAN_SDK env var is used only on windows
2/ as vulkan cmake module and target are required to pass qt configuration, maybe the best solution is to make `vulkan_-oader` a build_requirement instead of a requirement ?

Reply:
Or a private dependency, so that it doesn't contribute to PATH, LD_LIBRARY_PATH etc. 

Well it was jut an attempt to have better granularity on vulkan requirements. I think it's fine like this and could be improved later with vulkan recipes improvements.

But for macos, moltenvk is missing.